### ISSUE TITLE
 Partly deserializing a JsonObject for a known property

### ISSUE THREAD

**Issue by smuellener** [bblanchon/ArduinoJson#959]
I would like to deserialize a JsonObject partly to save memory by not deserializing properties of unknown size. 

Example:

I have a string containing a serialized partly known object:

```
{
    "UnknownProperty": "valueWithUnknownSize",
    "KnownProperty": "valueWithKnownSize"
}
```

Is it possible to only deserialize a known specific property from a serialized object?

E.g.:

```
StaticJsonDocument<64> doc;
deserializeJsonSpecificProperty(doc, example, "KnownProperty");
```

This would be helpful for a library that only knows (the required size for deserializing) part of the object.  Thanks!

**Comment by bblanchon** [comment]
Hi @smuellener,

I've thought about such a feature for a long time, but I don't see that coming in the near future.

For this feature to be complete, it would need to support nested members, arrays, multiple selectors, etc.
In the end, it would probably require a complete JSONPath implementation.
It's not impossible, but I'm concerned with the impact on code size and performance.
Is it something we want to do on a microcontroller? I'm not sure.

Anyways, I'll keep this issue opened as a feature request, but it will probably stay there for a while.

Meanwhile, you can work around this issue by skipping the part of the input you don't care.
Have a look at [How to deserialize a very large document?](https://arduinojson.org/v6/faq/how-to-deserialize-a-very-large-document/) for explanations.

Regards,
Benoit

**Comment by smuellener** [comment]
Very nice solution thanks!

**Comment by bblanchon** [comment]
`DeserilizationOption::Filter()` is more complicated than the solution you proposed, but it's also more powerful because you can select multiple members.

**Comment by bblanchon** [comment]
This feature was included in [ArduinoJson 6.15.0](https://arduinojson.org/news/2020/03/22/version-6-15-0/).
